src/scaling.jl

```diff
@@ -0,0 +1,89 @@
+"""
+    Scaling <: Transform
+
+Linearly scale the data as `ax + b`, according to some statistics `a` and `b`.
+"""
+abstract type Scaling <: Transform end
+
+"""
+    MeanStdScaling(mean, std) <: Scaling
+
+Linearly scale the data by a statistical `mean` and standard deviation `std`.
+This is also known as standardization, or the Z score transform.
+Once computed, the statistics of a `MeanStdScaling` are immutable.
+
+Can take a precomputed `mean` and `std` as arguments, or compute them from data.
+
+# Arguments
+* `mean::NamedTuple`: tuple of mean values, named by the scope of values it applies to.
+  `(all=μ, )` will apply to all data; `(1=μ1, 2=μ2)` for `AbstractArray` data will apply μ1
+  to the first slice and μ2 to the second slice; `(a=μ1, b=μ2)` for `Table` data will apply
+  μ1 to column `a` and μ2 to column `b`.
+* `std::NamedTuple`: similar to `mean` but for standard deviation values.
+
+# Keyword arguments to `apply`
+* `inverse=true`: inverts the scaling (e.g. to reconstruct the unscaled data)
+* `eps=1e-3`: replaces all 0 values in `std` before scaling (if `inverse=false`)
+"""
+struct MeanStdScaling <: Scaling
+    mean::NamedTuple
+    std::NamedTuple
+end
+
+"""
+    MeanStdScaling(data; kwargs...) <: Scaling
+
+Construct a [`MeanStdScaling`](@ref) using the mean and standard deviation of the data.
+
+# Keyword arguments
+* `dims=:`: for `AbstractArray` data, the dimension(s) to compute statistics along.
+* `cols=nothing`: for `Table` data, the column names to compute statistics for.
```

Comment:
Can you add a warning or note that dims and cols just be specified the same way when constructing and applying or else the resulsts will be inconsistent?